### ISSUE TITLE
Warning compiling h264.c for freerdp 2.0.0 rc2

### ISSUE THREAD

**Issue by weberhofer** [FreeRDP/FreeRDP#4573]
```
cd libfreerdp && /usr/bin/cc -DCMAKE_BUILD_TYPE=RelWithDebInfo -DEXT_PATH=\"/usr/lib64/freerdp2/extensions\" -DFREERDP_EXPORTS -DHAVE_CONFIG_H -DWINPR_EXPORTS -DWITH_OPENSSL -DWITH_WAYLAND -DWITH_X11 -DWITH_XKBFILE -D_FILE_OFFSET_BITS=64 -Dfreerdp_EXPORTS -I/home/abuild/rpmbuild/BUILD/FreeRDP-2.0.0-rc2/build -I/home/abuild/rpmbuild/BUILD/FreeRDP-2.0.0-rc2/build/include -I/home/abuild/rpmbuild/BUILD/FreeRDP-2.0.0-rc2/include -I/home/abuild/rpmbuild/BUILD/FreeRDP-2.0.0-rc2/winpr/include -I/home/abuild/rpmbuild/BUILD/FreeRDP-2.0.0-rc2/build/winpr/include  -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -fPIE -pie -DNDEBUG -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -g -O2 -g -DNDEBUG -Wno-builtin-macro-redefined -D__FILE__='"libfreerdp/codec/h264.c"' -fPIC   -o CMakeFiles/freerdp.dir/codec/h264.c.o   -c libfreerdp/codec/h264.c

libfreerdp/codec/h264.c: In function 'avc420_compress':
libfreerdp/codec/h264.c:219:41: warning: passing argument 2 of 'h264->subsystem->Compress' from incompatible pointer type [-Wincompatible-pointer-types]
   return h264->subsystem->Compress(h264, h264->pYUVData, h264->iStride, ppDstData, pDstSize);
                                          ^~~~
libfreerdp/codec/h264.c:219:41: note: expected 'const BYTE ** {aka const unsigned char **}' but argument is of type 'BYTE ** {aka unsigned char **}'
libfreerdp/codec/h264.c: In function 'avc444_compress':
libfreerdp/codec/h264.c:267:38: warning: passing argument 2 of 'h264->subsystem->Compress' from incompatible pointer type [-Wincompatible-pointer-types]
   if (h264->subsystem->Compress(h264, h264->pYUV444Data, h264->iStride, &coded, &codedSize) < 0)
                                       ^~~~
libfreerdp/codec/h264.c:267:38: note: expected 'const BYTE ** {aka const unsigned char **}' but argument is of type 'BYTE ** {aka unsigned char **}'
/home/abuild/rpmbuild/BUILD/FreeRDP-2.0.0-rc2/[-Wincompatible-pointer-types]
   if (h264->subsystem->Compress(h264, h264->pYUVData, h264->iStride, &coded, &codedSize) < 0)
                                       ^~~~
libfreerdp/codec/h264.c:274:38: note: expected 'const BYTE ** {aka const unsigned char **}' but argument is of type 'BYTE ** {aka unsigned char **}'

**Comment by akallabeth** [comment]
@weberhofer those are warnings, why can't you compile?

**Comment by weberhofer** [comment]
You are right, sorry. That's another issue - please close if you think it should be closed.

**Comment by weberhofer** [comment]
This warnings still occur in 2.0.0-rc3

**Comment by akallabeth** [comment]
Fixed with #4828
